liwords-ui/src/store/socket_handlers.ts

```diff
@@ -454,7 +456,7 @@ export const useOnSocketMsg = () => {
               payload: '',
             });
             const gid = nge.getGameId();
-            setRedirGame(gid);
+            historyRef.current.replace(`/game/${encodeURIComponent(gid)}`);
```

Comment:
could you explain why this instead of `history.replace`? I am a little confused about what the refs do here

Reply:
using ref avoids having to make `history` a dependency of this `useCallback`. here `historyRef.current` will refer to the latest observed instance of `history` because we reassign it every render.